tools/authorPatchCollector.py

```diff
@@ -0,0 +1,109 @@
+import argparse
+import subprocess
+
+
+def patch_collector(dir_path, authors, output):
+    print(f'\n**Inside {dir_path}**\n')
+    end = ''
+    begin = ''
+    patch_count = 1
+    range_count = 1
+
+    author_list = ''
+    for author in authors:
+        author_list += f'--author={author} '
+
+    get_hashes_cmd = (f'git -C {dir_path} log --pretty=format:"%h" '
+                      f'{author_list}')
+    hashes = subprocess.check_output(get_hashes_cmd, shell=True)
+    hashes = hashes.decode(encoding='utf-8')
+    hashes = hashes.split('\n')
+
+    for hash in hashes:
+
+        ancestor_hash_cmd = (f'git -C {dir_path} log --pretty=format:"%h" '
+                             f'{author_list}{hash}^^..{hash}^ --exit-code'
+                             ' 1>/dev/null')
+
+        if subprocess.Popen(ancestor_hash_cmd, shell=True).wait():
+            if end == "":
+                end = hash
+            begin = hash
+            range_count = range_count + 1
+
+        else:
+            subdir_msg = ('\nPlease enter a subdirectory name for this patch'
+                          '(leave this empty\nto leave this patch out, use ./ '
+                          'for the root directory):')
+            print('\n')
+
+            if end == "":
+                temp = patch_count-1
+                print(f'Adding one patch to {temp} existing:')
+                subprocess.Popen(
+                    f'git -C {dir_path} log {hash}^..{hash}', shell=True) \
+                    .wait()
+                print(subdir_msg)
+                subdir = input()
+
+                if subdir != '':
+                    subprocess.Popen(
+                        f'mkdir -p {output}/{subdir}', shell=True).wait()
+                    subprocess.Popen(
+                        f'git -C {dir_path} format-patch {hash}^..{hash} -o '
+                        f'{output}/{subdir}', shell=True).wait()
+                    patch_count = patch_count + 1
+
+                else:
+                    print('Omitting patch...')
+
+            else:
+                temp = patch_count-1
+                print(f'Adding {range_count} patches to {temp} existing')
+                subprocess.Popen(
+                    f'git -C {dir_path} log {begin}^^..{end} --oneline',
+                    shell=True).wait()
+                print(subdir_msg)
+                subdir = input()
+
+                if subdir != '':
+                    print(f'mkdir -p {output}/{subdir}')
+                    subprocess.Popen(
+                        f'mkdir -p {output}/{subdir}', shell=True).wait()
+                    subprocess.Popen(
+                        f'git -C {dir_path} format-patch {begin}^^..{end} '
+                        f'-o{output}/{subdir}', shell=True).wait()
+                    patch_count = patch_count + range_count
+
+                else:
+                    print('Omitting patch series...')
+
+                end = ""
```

Comment:
this use of `end` and `start` feels like they are booleans, or loop variables, and shouldnt be strings.

Reply:
They're used for preparing the patches, line 64 and 75.

Reply:
ah I see, then again give them meaningful names , at least `foo_commit` or `foo_sha`

Reply:
`begin_hash` and `end_hash` would suffice, I guess?